query/parser.go

```diff
@@ -99,57 +99,60 @@ var dateFormats = []string{
 	time.RFC822Z,
 }
 
-// `now` isn't captured in this regex
-var relativeTimeRegexp = regexp.MustCompile(`^-(\d+)([a-zA-Z])$`)
+var (
+	relativeTimeRegexp      = regexp.MustCompile(`^now\s*([+-])\s*([0-9]+)([smhdwMy])$`)
+	relativeTimeRegexpShort = regexp.MustCompile(`(-)\s*([0-9]+)([smhdwMy])$`)
+)
+
+// parseDate converts the given datestring (from one of the allowable formats) into a millisecond offset from the Unix epoch.
+func parseDate(date string, now time.Time) (int64, error) {
```

Comment:
If you keep the relative time conversion as a separate function, you can reuse it for parsing the `resolution` duration.